### PR TITLE
Fixed overflow and added corner case tests

### PR DESCRIPTION
**Pull request by xgreenx**
Close https://github.com/FuelLabs/fuel-core/issues/763

### CODE REVIEW

File: fuel-core/src/resource_query.rs [hunk 1]
```diff
@@ -734,6 +741,104 @@ mod tests {
...
+        ; "Enough resources in the DB to reach target(u64::MAX) by 1 resource"
+    )]
+    #[test_case::test_case(
+        TestCase {
+            db_amount: vec![1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, u64::MAX],
+            target_amount: u64::MAX,
+            max_resources: 1,
+        }
+        => Ok(1)
+        ; "Enough resources in the DB to reach target(u64::MAX) by two resources"
```
Comment (Voxelot) [inline fuel-core/src/resource_query.rs:778]:
```suggestion
        ; "Enough resources in the DB to reach target(u64::MAX) by 1 resource"
```

**Review by Voxelot — APPROVED** [review]

**Review by Voxelot — APPROVED** [review]

**Review by Voxelot — APPROVED** [review]
